packages/react-web-worker/src/hooks.ts

```diff
@@ -0,0 +1,19 @@
+import {useEffect} from 'react';
+// import {terminate} from '@shopify/web-worker';
+
+import {useLazyRef} from '@shopify/react-hooks';
+
+export function useWorker<T>(creator: () => T) {
+  const {current: worker} = useLazyRef(creator);
+
+  useEffect(
+    () => {
+      return () => {
+        // terminate(worker);
```

Comment:
Will add in once #1111 is merged.
I'm ok with shipping this independently from #1111